File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int n = nums.length;
        ArrayList<Integer> pos= new ArrayList<>();
        ArrayList<Integer> neg = new ArrayList<>();

        int i =0;
        while(i<n){
            if(nums[i]<0){
                neg.add(nums[i]);
            }
            else{
                pos.add(nums[i]);
            }
            i++;
        }
       

       int p =0;
       int q= 0;

       for(int j= 0;j<n;j++){
        if(j%2==0){
            nums[j] = pos.get(p);
            p++;
        }
        else{
            nums[j]= neg.get(q);
            q++;
        }
       }
       return nums;
    }
}

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        pos = []
        neg = []
        for i in nums:
            if i<0:
                neg.append(i)
            else:
                pos.append(i)    
        answer = [0]*n 
        p = 0
        q = 0
        for k in range(n):
            if k%2== 0:
                answer[k] = pos[p]
                p+=1
            else:
                answer[k] = neg[q]
                q+=1
        return answer               